olive/engine/engine.py

```diff
@@ -861,6 +863,9 @@ def _run_pass(
                 input_model = self._prepare_non_local_model(input_model)
             try:
                 output_model = host.run_pass(p, input_model, output_model_path, pass_search_point)
+            except (ValueError, TypeError, AttributeError):
+                # Don't catch these errors since most of time, it is caused by the user errors and need not retry.
+                raise
             except Exception:
                 output_model = PRUNED_CONFIG
                 # TODO: from the time being, we need to catch all exceptions to make the
```

Comment:
Nit: Shall we remove this comments? As you already fix it.

Reply:
actually, it is still need some works to make sure the exception handling fully done. So I keep the TODO here